frontend/public/components/edit-yaml.jsx

```diff
@@ -643,21 +647,61 @@ export const EditYAML = connect(stateToProps)(
       });
     }
 
+    toggleSidebar = () => {
+      this.setState((state) => {
+        return { showSidebar: !state.showSidebar };
+      });
+      window.dispatchEvent(new Event('sidebar_toggle'));
+    };
+
     render() {
       if (!this.props.create && !this.props.obj) {
         return <Loading />;
       }
 
-      const { connectDropTarget, isOver, canDrop } = this.props;
+      const { connectDropTarget, isOver, canDrop, create, yamlSamplesList } = this.props;
       const klass = classNames('co-file-dropzone-container', {
         'co-file-dropzone--drop-over': isOver,
       });
 
-      const { error, success, stale, yaml, height } = this.state;
-      const { create, obj, download = true, header } = this.props;
+      const { error, success, stale, yaml, height, showSidebar } = this.state;
+      const { obj, download = true, header } = this.props;
       const readOnly = this.props.readOnly || this.state.notAllowed;
       const options = { readOnly, scrollBeyondLastLine: false };
       const model = this.getModel(obj);
+      const { samples, snippets } = getResourceSidebarSamples(model, yamlSamplesList);
+      const showSamples = !_.isEmpty(samples) && create;
+      const sidebarLinkLabel = `View Schema ${showSamples ? 'and Samples' : ''}`;
```

Comment:
This seems like a good time to fix https://jira.coreos.com/browse/CONSOLE-1611 since we're adjusting the logic here. We should check if a schema is available as well and hide the sidebar if not.